### PR TITLE
Feature/r900mini rx

### PR DESCRIPTION
**Pull request by dimitroffs**
Hello fellows! Yesterday I managed to trace (as far as I could in the late hours) the pinout of Jumper's R900 mini rx. Here are my first impressions and thoughts:
- rx is small size and cheap, suitable with Jumper's R9M implementation (MPM) and FrSky's R9M systems
- as you can check on detailed pictures, there are no additional pins (comfortable to solder), except Vdd, Gnd and Sbus.
- all default pins of R9MINI_RX match
- led and btn pins match R9Slim+ pinout (didn't checked if leds match in color, but in worst case they will be reversed, so we are going to correct that)
- no rx/tx pins comfortable for solder - I managed to find tx1 on one of the chips so it will be good for solder, but in the code I selected both tx2/rx2 of stmf103 due to the fact they are end corner pins of the chip itself and comfortable situated on the one side of the rx pcb.
- I created basic build for platformio (copied by r9slim+ pr)

Check the pictures and lets discuss how to proceed next. Hope anyone else could try also, but in worst case I have 3 spares which can burn for the good of the project 🔥 

Hope this helps!

### CODE REVIEW

File: src/user_defines.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@
 
 -DAUTO_WIFI_ON_BOOT
 
-#-DUSE_ESP8266_BACKPACK
+-DUSE_ESP8266_BACKPACK
```
Comment (cruwaller) [inline src/user_defines.txt:54]: This should not be changed. Please revert this.

File: src/src/targets.h [hunk 2]
```diff
@@ -136,6 +139,11 @@ Credit to Jacob Walser (jaxxzer) for the pinout!!!
     /* PB9: antenna 1 (left) = HIGH, antenna 2 (right) = LOW
      * Note: Right Antenna is selected by default, LOW */
     #define GPIO_PIN_ANTENNA_SELECT PB9
+#elif TARGET_R900MINI_RX
```
Comment (cruwaller) [inline src/src/targets.h:142]: Wow, this uses same fw with R9SLIM+ 😄👌

**Review by cruwaller — CHANGES_REQUESTED** [review]

**Review by cruwaller — APPROVED** [review]
Let's try to find better solution for communication later :)
This is good starting point! Thanks!

File: src/platformio.ini [hunk 3]
```diff
@@ -189,6 +189,58 @@ lib_deps =
...
+build_unflags = -Os
+build_flags =
+	-D TARGET_R9M_RX
+	-D TARGET_R900MINI_RX
+	-D TARGET_100mW_MODULE
+	-D PLATFORM_STM32
+	-D HSE_VALUE=24000000U
+	-O2
+	-DVECT_TAB_OFFSET=0x08008000U
+board_build.ldscript = variants/R9MM/R9MM_ldscript.ld
```
Comment (dimitroffs) [inline src/platformio.ini:205]: not sure if we need to change this, I am not familiar with those scripts

Comment (cruwaller) [reply src/platformio.ini:205]: No need to duplicate it just for different naming. These linker files should be merged to one later.

**Review by cruwaller — APPROVED** [review]

File: src/platformio.ini (hunk 3 above)
Comment (dimitroffs) [reply src/platformio.ini:205]: yeah, I saw on some targets they use inheritance from others, so probably it will be reworked at some point

**Comment by AlessandroAU** [comment]
let me know when this is ready 👍

**Comment by dimitroffs** [comment]
> let me know when this is ready 👍 

Pete waits for his R900mini rx and will confirm if all this stuff is ready for action. I successfully flashed one of mine - works as expected.. will just finally test stlink upload via pio, coz this is what I haven't tried already, but we actually flashed the rx bootloader via stlink (manually) so it should be good to go. Will confirm it today.

**Comment by dimitroffs** [comment]
I managed to try stlink update - it works as expected - flashed bootloader successfully. Passthrough works also, so when you have time to check it out on your side - lets merge this :) I corrected build scripts as others, hope I didn't crash something :D Regards!

**Pull request merged**